src/main/java/com/google/devtools/build/lib/skyframe/RegisteredExecutionPlatformsFunction.java

```diff
@@ -234,7 +234,10 @@ public boolean shouldRetain(Target target, boolean explicit) {
       if (explicit) {
         return true;
       }
+      return hasPlatformInfo(target);
+    }
 
+    boolean hasPlatformInfo(Target target) {
```

Comment:
The owning class for this method is very strange - it's an AutoValue without fields? (Documentation would help!)

In any case this method should likely be static now and possibly live elsewhere so it can be used from here and its new usage properly.

Reply:
Moved and cleaned up the comment.

Reply:
The AutoValue is present to give hashCode and equals implementations (needed for AutoCodec to work properly), and a toString. I can remove it and add simple versions of those, which is better?